packages/hoppscotch-common/src/components/collections/Collection.vue

```diff
@@ -267,16 +289,18 @@ const emit = defineEmits<{
   (event: "dragging", payload: boolean): void
   (event: "update-collection-order", payload: DataTransfer): void
   (event: "update-last-collection-order", payload: DataTransfer): void
+  (event: "run-collection", collectionID: string): void
 }>()
 
-const tippyActions = ref<TippyComponent | null>(null)
-const requestAction = ref<HTMLButtonElement | null>(null)
-const folderAction = ref<HTMLButtonElement | null>(null)
-const edit = ref<HTMLButtonElement | null>(null)
-const deleteAction = ref<HTMLButtonElement | null>(null)
-const exportAction = ref<HTMLButtonElement | null>(null)
-const options = ref<TippyComponent | null>(null)
-const propertiesAction = ref<TippyComponent | null>(null)
+const tippyActions = ref<VNodeRef | null>(null)
```

Comment:
what is the reason behind using `VNodeRef` here ? It includes very generic values like 

1. `Ref<any>` which will match any ref
2. string which will match any string

so typescript will not catch assignments like this,
```typescript
tippyActions.value = "whatever" // no TS error
``` 

look into the vue docs, on their recommendation for typing refs.
[https://vuejs.org/guide/typescript/composition-api#typing-template-refs](https://vuejs.org/guide/typescript/composition-api#typing-template-refs)

Reply:
Noticed a few type errors arising in the markup with a few property access via the template ref; picked the inferred type from the `ref` attribute. Yeah, better to narrow it down; push an [update](https://github.com/hoppscotch/hoppscotch/pull/4141/commits/9e1a454f8c76877b261a4a31bd35031cef34cebd).